.clang-format

```diff
@@ -54,7 +54,7 @@ UseTab: Never
 IncludeBlocks: Regroup
 IncludeCategories:
 # first our config files:
-  - Regex: "(IBTK|IBAMR)_config.h"
```

Comment:
Do we have compatibility headers named `IBTK_config.h` and `IBAMR_config.h`? Could this cause problems on OSes that do not use case sensitive file systems by default (e.g. MacOS)?

Reply:
Oops, I misread this --- the new header is `ibtk/config.h` --- so it should be fine.

Reply:
They still exist (and will exist as long as the autotools build system is around) but they are not in the CMake build system (these top-level config files don't work with `make install`). I removed them here since nothing in the library will include them once I am done with this round of patches.